### DISCUSSION TITLE
Heroku deployment failed - can't resolve absolute Imports and module path aliases

### DISCUSSION THREAD

**Discussion by tttoh** [vercel/next.js#22521] [Help]
I have implemented absolute import as per documentation here: https://nextjs.org/docs/advanced-features/module-path-aliases

environment setup
```
"next": "10.0.7",
"react": "17.0.1",
"react-dom": "17.0.1",
"serverless-mysql": "^1.5.4",
"swr": "^0.4.2"
```
when npm run build locally, it run successfully without an issue but failed to deploy in heroku.
```
jsconfig.json
{
    "compilerOptions": {
      "baseUrl": ".",
      "paths": {
        "@/components/*": ["components/*"],
        "@/helper/*": ["helper/*"],
        "@/scripts/*": ["scripts/*"],
        "@/lib/*": ["lib/*"],
        "@/pages/*": ["pages/*"]
      }
    }
  }
```
Heroku build Log
```
Failed to compile.
ModuleNotFoundError: Module not found: Error: Can't resolve '@/components/Layout' in '/tmp/build_96463f76_/pages'
> Build error occurred
Error: > Build failed because of webpack errors
    at /tmp/build_96463f76_/node_modules/next/dist/build/index.js:15:918
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /tmp/build_96463f76_/node_modules/next/dist/build/tracer.js:3:470
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! next-myapp@0.1.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the next-myapp@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/npmcache.QqjQC/_logs/2021-02-25T08_04_01_637Z-debug.log
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Node version not specified in package.json
         https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
       
 !     Push rejected, failed to compile Node.js app.
 !     Push failed
```

Anyone facing this issue or know the solution?  Thanks you.

Solved:

Case sensitivity issue. It is regarding using import path with uppercase `Layout` instead of lowercase in directory.
Somehow it still works in my Windows development environment and not throwing any warning or error.

**Comment by RyanMan56** [comment]
> Solved:
>
> Case sensitivity issue. It is regarding using import path with uppercase Layout instead of lowercase in directory.
> Somehow it still works in my Windows development environment and not throwing any warning or error.


Had the same problem, this was 100% the issue. If anyone else has this problem, check the casing of the files in your git repo online matches what you're seeing locally

**Comment by conor909** [comment]
I have a next.js app and followed [this quick tutorial](https://docs.amplify.aws/start/getting-started/hosting/q/integration/next/#set-up-cicd-for-your-app) to install amplify, for some reason when deploying to heroku it fails and it says a file cannot be found.  It's there, in the place specified, I've checked casing and paths, and it works perfect locally.  But when deploying to heroku it says the module is not found.

```js
remote: ./src/pages/app.js
remote: Module not found: Can't resolve '../aws-exports' in '/tmp/build_c9a09900/src/pages'
```

my folder structure is as follows:
```js
src
L pages
  L index.js 
  L app.js
L aws-exports.js
```

app.js
```js
import Head from 'next/head'
import Image from 'next/image'
import styles from '../../styles/Home.module.css'
import { Amplify } from 'aws-amplify'
import awsExports from '../aws-exports'
import { withAuthenticator, Button, Heading } from '@aws-amplify/ui-react'
import '@aws-amplify/ui-react/styles.css'

Amplify.configure(awsExports)

function App({ signOut, user }) {
  return (
    <>
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Heading level={1}>Hello {user.username}</Heading>
      <Button onClick={signOut}>Sign out</Button>
    </div>
    </>
  )
}

export default withAuthenticator(App)

```

**Reply by GeorgeDubuque** [reply]
any update on this one, would love to know what was going wrong here! :)

**Comment by inapeace0** [comment]
I got the same issue when I deploy next.js app on heroku. **Any solution to solve it?**

**Comment by MaeAERIAH369** [comment]
Same issue

**Comment by JamesScript** [comment]
As suggested [earlier](https://github.com/vercel/next.js/discussions/22521#discussioncomment-454893) it turns out it was casing for me too. I had to check this twice as I was convinced the casing was the same the first time, but turned out I was trying to import an icon named "FileCsv.svg" when it was actually called "FileCSV.svg".
